src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/unit_tests/OrderlyServerTests.kt

```diff
@@ -129,15 +129,72 @@ class OrderlyServerTests: TeamcityTests()
         testTranslatedResponse(json, json)
     }
 
-    private fun testTranslatedResponse(rawResponse: String, expectedTranslatedResponse: String) {
+    @Test
+    fun `throws on failure if specified`()
+    {
+        val text = """{"status": "failure", "data": null, "errors": []}"""
+        val rawJson = JSONObject(text)
+        val mockRawResponse = mock<Response> {
+            on { this.jsonObject } doReturn rawJson
+            on { this.statusCode } doReturn 400
+        }
+
+        val mockClient = mock<HttpClient> {
+            on { this.get(any(), any()) } doReturn mockRawResponse
+        }
+
+        val sut = OrderlyServer(mockConfig, mockClient)
+                .throwOnError()
```

Comment:
I don't think I've seen this kind of pattern before, what is it called?

Reply:
it's kind of a fluent builder pattern